src/speechly/client.ts

```diff
@@ -211,27 +217,43 @@ export class Client {
   }
 
   /**
-   * Stops current SLU context by sending a stop context event to the API and muting the microphone.
+   * Stops current SLU context by sending a stop context event to the API and muting the microphone
+   * delayed by contextStopDelay = 250 ms
    */
   async stopContext(): Promise<string> {
     if (this.state !== ClientState.Recording) {
       throw Error('Cannot stop context - client is not recording')
     }
 
     this.setState(ClientState.Stopping)
-    this.microphone.mute()
+    const contextId: string = await new Promise((resolve) => {
+      Promise.race([
+        new Promise((resolve) => setTimeout(resolve, this.contextStopDelay)), // timeout
+        new Promise((resolve) => { this.resolveStopContext = resolve }),
+      ])
+        .then(() => {
```

Comment:
Let's use async / await syntax here. Mixing different syntax styles makes the code harder to read and async / await produces more readable code in my mind.

Reply:
We already mixing the styles, for example we create promises in WebsocketClient. We need to store the resolver to force the stopContext before timeout end, thus Promise syntax needed here.